pkg/controllers/jobset_controller.go

```diff
@@ -410,8 +444,19 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 			}
 			log.V(2).Info("successfully created job", "job", klog.KObj(job))
 		})
+		if !jobSetSuspended(js) && inOrderStartupPolicy(startupPolicy) {
+			return r.ensureCondition(ctx, js, corev1.EventTypeNormal, true, generateStartupPolicyCondition(false))
```

Comment:
I think ```ensureCondition``` is a good candidate to be refactored to accept an option struct.

This is because:
- The number of unnamed parameters is growing too long (5) 
- I am not a fan of raw boolean parameters for functions whose names do not make it explicitly clear what the semantic meaning of the boolean value is.

This also applies to functions with fewer parameters like: `generateStartupPolicyCondition(false)`; it is unclear what the true/false signifies, and you have to go look at the function definition to determine that, then go back to the original code and start reading again with that in mind. 

This can be improved by accepting a option struct, allowing for named parameters:

```go
startupPolicyCondition := generateStartupPolicyCondition(&startupPolicyConditionOpts{
  status: metav1.ConditionFalse,
  message: "startup policy in order starting",
})

return r.ensureCondition(ctx, &ensureConditionOpts{
  jobSet: js,
  eventType: metav1.EventTypeNormal,
  forceFalseUpdate: true,
  condition: startupPolicyCondition,
})

```


We have a few functions throughout the codebase whose readability could be improved with option structs (or other changes), we don't need to address them all here, but if you agree this is a good change we should start here.

Reply:
Pushed a refactor for ensureConditions that sorta follows what you suggested. PTAL.

Reply:
Updated